src/lsptoolshost/copilot.ts

```diff
@@ -26,74 +27,129 @@ interface CopilotRelatedFilesProviderRegistration {
             cancellationToken?: vscode.CancellationToken
         ) => Promise<{ entries: vscode.Uri[]; traits?: CopilotTrait[] }>
     ): vscode.Disposable;
+    getContextProviderAPI(version: string): Promise<ContextProviderApiV1 | undefined>;
 }
 
-export function registerCopilotExtension(languageServer: RoslynLanguageServer, channel: vscode.LogOutputChannel) {
+async function getCopilotAPIsAsync(): Promise<CopilotAPIs | undefined> {
     const ext = vscode.extensions.getExtension('github.copilot');
     if (!ext) {
-        channel.debug('GitHub Copilot extension not installed. Skip registeration of C# related files provider.');
-        return;
+        return undefined;
     }
-    ext.activate().then(() => {
-        const relatedAPI = ext.exports as CopilotRelatedFilesProviderRegistration | undefined;
-        if (!relatedAPI) {
-            channel.debug(
-                'Incompatible GitHub Copilot extension installed. Skip registeration of C# related files provider.'
-            );
-            return;
-        }
 
-        channel.debug('registration of C# related files provider for GitHub Copilot extension succeeded.');
+    if (!ext.isActive) {
+        try {
+            return await ext.activate();
+        } catch {
+            return undefined;
+        }
+    } else {
+        return ext.exports as CopilotAPIs | undefined;
```

Comment:
guessing this fails if the interface definition doesn't match?  e.g. the version of the extension doesn't have a `getContextProviderAPI` export?